tests/Unit/Evolution/DiscontinuousGalerkin/Actions/Test_NormalCovectorAndMagnitude.cpp

```diff
@@ -0,0 +1,306 @@
+      normal_covector_quantities{};
+  for (const auto& [direction, local_neighbors] : element.neighbors()) {
+    (void)local_neighbors;
+    normal_covector_quantities[direction] = std::nullopt;
+  }
+
+  return db::create<simple_tags, compute_tags>(
+      1.0,
+      Mesh<Dim>{number_of_grid_points_per_dim, Spectral::Basis::Legendre,
+                Spectral::Quadrature::Gauss},
+      std::move(grid_to_inertial_map), std::move(element),
+      std::move(normal_covector_quantities));
+}
+
+template <bool UseFlatSpace, size_t Dim, typename DbTagsList>
+void check_normal_covector_quantities(
+    const gsl::not_null<db::DataBox<DbTagsList>*> box) noexcept {
+  using field_face_tags = tmpl::conditional_t<
+      UseFlatSpace,
+      tmpl::list<evolution::dg::Actions::detail::OneOverNormalVectorMagnitude>,
+      tmpl::list<evolution::dg::Actions::detail::OneOverNormalVectorMagnitude,
+                 evolution::dg::Actions::detail::NormalVector<Dim>,
+                 InverseSpatialMetric<Dim>>>;
+  using internal_directions = domain::Tags::InternalDirections<Dim>;
+  const auto& unnormalized_normal_covectors = db::get<domain::Tags::Interface<
+      internal_directions, domain::Tags::UnnormalizedFaceNormal<Dim>>>(*box);
+  for (const auto& [direction, neighbors] :
+       db::get<domain::Tags::Element<Dim>>(*box).neighbors()) {
+    (void)neighbors;
+    Variables<field_face_tags> fields_on_face{
+        db::get<domain::Tags::Interface<internal_directions,
+                                        domain::Tags::Mesh<Dim - 1>>>(*box)
+            .at(direction)
+            .number_of_grid_points()};
+    if constexpr (not UseFlatSpace) {
+      double temp = 0.1;
+      for (auto& component : get<InverseSpatialMetric<Dim>>(fields_on_face)) {
+        for (double& t : component) {
+          t = temp;
+          temp *= 1.1;
+        }
+      }
+    }
+
+    db::mutate<
+        evolution::dg::Tags::InternalFace::NormalCovectorAndMagnitude<Dim>>(
+        box,
+        &evolution::dg::Actions::detail::
+            unit_normal_vector_and_covector_and_magnitude<
+                tmpl::conditional_t<UseFlatSpace, FlatSpaceSystem,
+                                    SystemWithInverseMetric<Dim>>,
+                Dim, field_face_tags>,
+        make_not_null(&fields_on_face), direction,
+        unnormalized_normal_covectors,
+        db::get<domain::CoordinateMaps::Tags::CoordinateMap<Dim, Frame::Grid,
+                                                            Frame::Inertial>>(
+            *box));
+
+    if constexpr (UseFlatSpace) {
+      CHECK_ITERABLE_APPROX(
+          (DataVector{fields_on_face.number_of_grid_points(), 1.0}),
+          get(magnitude(
+              get<evolution::dg::Tags::InternalFace::NormalCovector<Dim>>(
+                  *get<evolution::dg::Tags::InternalFace::
+                           NormalCovectorAndMagnitude<Dim>>(*box)
+                       .at(direction)))));
```

Comment:
To make this easier to read, I suggest assigning the argument of `magnitude` to a variable `const auto& normal_covector = ...` as you've done below